tests/30rooms/13guestaccess.pl

```diff
@@ -311,97 +312,94 @@
    check => sub {
       my ( $http, $user ) = @_;
 
-      Future->needs_all(
-         matrix_create_room( $user,
-            visibility => "public",
-            room_alias_name => "listingtest0",
-         ),
+      my %rooms = (
+         listingtest0 => {},
+         listingtest1 => { history_visibility => "world_readable" },
+         listingtest2 => { history_visibility => "invited" },
+         listingtest3 => { guest_access => "can_join" },
+         listingtest4 => { guest_access => "can_join", history_visibility => "world_readable" },
+      );
 
-         matrix_create_room( $user,
-            visibility => "public",
-            room_alias_name => "listingtest1",
-         )->then( sub {
-            my ( $room_id ) = @_;
+      Future->needs_all( map {
+         my $aliasname = $_;
+         my $settings = $rooms{$_};
+         my $room_id;
 
-            matrix_set_room_history_visibility( $user, $room_id, "world_readable" );
-         }),
-
-         matrix_create_room( $user,
+         my $f = matrix_create_room( $user,
             visibility => "public",
-            room_alias_name => "listingtest2",
-         )->then( sub {
-            my ( $room_id ) = @_;
-
-            matrix_set_room_history_visibility( $user, $room_id, "invited" );
-         }),
-
-         matrix_create_room( $user,
-            visibility => "public",
-            room_alias_name => "listingtest3",
-         )->then( sub {
-            my ( $room_id ) = @_;
+            room_alias_name => $aliasname,
+         )->on_done( sub {
+            ( $room_id, undef ) = ( @_ );
+         });
 
-            matrix_set_room_guest_access( $user, $room_id, "can_join" );
-         }),
+         if( $settings->{guest_access} ) {
+            $f = $f->then( sub {
+               matrix_set_room_guest_access( $user, $room_id, $settings->{guest_access} );
```

Comment:
I'd have been tempted to use `initial_state` on `/createRoom` API

Reply:
fair. I was just sticking with the way the test did it originally, but with less C&P.